### DISCUSSION TITLE
filter .in

### DISCUSSION THREAD

**Discussion by EdgarsLv** [supabase/supabase#5683] [Questions]
If user have selected values then i want to filter user selected values,  if user have not selected any value i want to display all. 
Is this ok way to do that?

.in('column, some_array.length > 0 ? some_array  :  array_of_ALLVALUES)

And btw range filters are not working... documentation shows that have to pass array, but in code it asks for string.
Or i dont't understand there something?

![image](https://user-images.githubusercontent.com/56837461/155861134-4360acf9-4f1c-42cc-93ad-91eef182cb00.png)

**Comment by GaryAustin1** [comment]
Not really clear what your in statement is doing.
.in should look like this  .in('name', ['Rio de Janeiro', 'San Francisco'])  basically column name and array of possible values to match
If you want to not use the .in if the array is empty, then do something like the conditional chaining here to not include it in the query:
https://supabase.com/docs/reference/javascript/using-filters#conditional-chaining

I can't help with range filters, I don't have a handle on them either.

**Reply by EdgarsLv** [reply]
updated initial question to better understand

**Reply by GaryAustin1** [reply]
I'd just set your array before you query then.  query_array = some_array.length > 0 ? some_array : array_of_ALLVALUES
.in('column',query_array)....  but you have a typo with missing ' after column, so your logic might work with that fix.

edit: BUT that is not very efficient versus just not putting the .in filter with conditional logic like example linked when you want all data.

**Comment by krystian-dev548** [comment]
Maybe you can try this? https://supabase.com/docs/reference/javascript/using-filters

![Screenshot 2023-06-02 alle 11 42 48](https://github.com/supabase/supabase/assets/24765669/be265daf-aa68-432e-9567-e94345ca37f7)

**Comment by hvardhan907** [comment]
.select().gte('population',150).lte('population',250)
